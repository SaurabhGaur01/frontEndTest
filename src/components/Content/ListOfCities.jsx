import React from 'react';
import PropTypes from 'prop-types';
import { transformStrings, isAirportOrCity } from '../../utils/index';

const DOCS_KEY_FIELD = 'docs';
const CITY_BOOKING = 'city';
const AIRPORT_BOOKING = 'airport';

const ListOfCities = ({ listOfCities, keyFields, getSelectedCity }) => {
    if(keyFields === DOCS_KEY_FIELD) {
        return listOfCities[keyFields].map(item => {
            const { name, index, bookingId, city, region, country } = item;
            return (
                <li className="listingItem ml-0" key={index} onClick={() =>getSelectedCity(item)}>
                    <div className="d-flex pb-3">
                        {
                            isAirportOrCity(bookingId) === CITY_BOOKING && (
                                <span className="badge bg-primary mr-3 mt-2 iconDisplay">City</span>
                            )
                        }
                        {
                            isAirportOrCity(bookingId) === AIRPORT_BOOKING && (
                                <span className="badge bg-warning mr-3 mt-2 iconDisplay">Airport</span>
                            )
                        }
                        <div className="details">
                            <p className="m-0">{name}</p>
                            <p className="m-0">{transformStrings(city, region, country)}</p>
                        </div>
                    </div>
                </li> 
            )
        });
    }

    return null;
}

ListOfCities.propTypes = {
    keyFields: PropTypes.string.isRequired,
    getSelectedCity: PropTypes.func.isRequired,
    listOfCities: PropTypes.shape({
        docs: PropTypes.arrayOf(PropTypes.shape({
            name: PropTypes.string, 
            index: PropTypes.number, 
            bookingId: PropTypes.string,
            city: PropTypes.string,
            region: PropTypes.string,
            country: PropTypes.string, 
        })),
    }).isRequired,
}

export default ListOfCities;