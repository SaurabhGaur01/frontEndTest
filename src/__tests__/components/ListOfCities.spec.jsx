import React from 'react';
import { shallow } from 'enzyme';
import ListOfCities from '../../components/Content/ListOfCities';

describe('ListOfCities', () => {
    let props;

    beforeEach(() =>{
        props = {
            getSelectedCity: jest.fn(),
            listOfCities: {
                docs: [{
                    name: 'mockName',
                    index: 1,
                    bookingId: 'city-123',
                    city: 'mockCity',
                    region: 'mockRegion',
                    country: 'mockCountry',
                }],
            },
            keyFields: 'docs',
        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<ListOfCities  {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    it('should render correctly when keyFields is not docs', () => {
        props.keyFields = 'xyz';
        const renderedModule = shallow(<ListOfCities  {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });
})