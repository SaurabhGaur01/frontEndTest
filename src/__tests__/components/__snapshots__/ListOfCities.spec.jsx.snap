// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfCities should render correctly 1`] = `
<li
  className="listingItem ml-0"
  key="1"
  onClick={[Function]}
>
  <div
    className="d-flex pb-3"
  >
    <span
      className="badge bg-primary mr-3 mt-2 iconDisplay"
    >
      City
    </span>
    <div
      className="details"
    >
      <p
        className="m-0"
      >
        mockName
      </p>
      <p
        className="m-0"
      >
        mockCity, mockRegion, mockCountry
      </p>
    </div>
  </div>
</li>
`;

exports[`ListOfCities should render correctly when keyFields is not docs 1`] = `""`;
